### ISSUE TITLE
 NVDA, in Form's mode, the fields name are not announced, however it is displaying the same in the Speech Viewer

### ISSUE THREAD

**Issue by Olina-Roy** [nvaccess/nvda#7475]
[Contact Form_NVDA issue.docx](https://github.com/nvaccess/nvda/files/1200594/Contact.Form_NVDA.issue.docx)

In Form's mode, using a tab key the fields name are not announced in NVDA screen reader, however it is displaying in NVDA Speech Viewer.

### Steps to reproduce:
>When on the first field of the form, NVDA  is announcing the first field  as “First Name edit required blank”. 
>If you leave it blank and Tab to the next field, it is reading only the error of the previous blank field and not the Name,role of the present field, however in speech viewer you can see the field name (For screen shot please find the attached)
>
>Browser Tested on: Firefox
>Screen Reader version:2016 and 2017.2

**Comment by PratikP1** [comment]
I can confirm this behavior in Firefox and the latest next build. I was just about to report this. I have been able to replicate this on several forms. The behavior occurs if “speak typed words” is switched on but does not if it is switched off. Additionally, this behavior only happens using Microsoft OneCore voices. It does not occur if ESpeak NG or Codefactory Eloquence is selected. (I have attached a log but nothing much illuminating there from my perspective at least.) 😍  I am also providing detailed steps below.

Use the following steps to replicate.

STR:

  1.  Visit the following link in Firefox: http://clark.com/askclark/
  2.  Switch to Microsoft OneCore voices in the synthesizer preferences.
  3.  Switch on “speak typed words” by pressing NVDA+3 if it is switched off.
  4.  Go to the “first name” form field and switch off browse mode.
  5.  Type your name and press tab.
  6.  Fill out the rest of the form.
  7.  Switch off “speak typed words”  by pressing NVDA+3 and repeat steps 1 through 6.

Tested with Firefox 54.0.1 and NVDA version next-14285,ded94003.

**Comment by PratikP1** [comment]
@jcsteh, mentioning you based on our Twitter conversation.

**Comment by jcsteh** [comment]
Closing as a duplicate of #7473, since i believe it to be the same underlying issue.

**Comment by PratikP1** [comment]
@jcsteh You are correct. I can confirm with the try build from #7473<https://github.com/nvaccess/nvda/issues/7473> that this issue is resolved as well. Thank you.

**Comment by Olina-Roy** [comment]
@jcstech I installed and ran the copy, but issue still persists.Is there anything else which I am missing out on.

**Comment by jcsteh** [comment]
I think @Olina-Roy and @PratikP1 are referring to two different issues here. Looking more closely at @Olina-Roy's description, it mentions an error reported about the first field being blank, but this does not happen with the test case that @PratikP1 provided, regardless of the synthesizer.

@Olina-Roy, I think this is actually a duplicate of #3807 (On a Form that is enhanced with Aria invalid, NVDA does not read the next Formfield after announcing the error alert). Does this seem more like the issue you are experiencing? Also, please provide a URL demonstrating the issue you're experiencing; neither your description nor your attachment included a URL.

**Comment by PratikP1** [comment]
You're right. I did not read @Olina-Roy's original comment correctly. I conflated this with the issue I was seeing.

**Comment by Olina-Roy** [comment]
@jctech, yes! #3807 is more like what I have been experiencing.Additionally, in the speech viewer I can see the currently focused field getting declared before the error message for the previous field, but its not getting read out.

The page is still under development, so there is no production url which I could share to demonstrate the issue. :(

**Comment by Olina-Roy** [comment]
I have added the screenshot in the document attached on my Original Post.

**Comment by jcsteh** [comment]
Closing as duplicate of #3807.
